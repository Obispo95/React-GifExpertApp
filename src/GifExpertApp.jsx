import { useState } from "react"
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['Cats']);

  const onAddCategory = (newCategory) => {

    // setCategories(categories => [...categories, 'Valorant']);
    // setCategories(['Valorant', ...categories]);
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  }

  return (
    <>
      <center>
        <h1 className="title">GIF EXPERT APP</h1>

        <AddCategory
          onNewCategory={onAddCategory}
        />

        {
          categories.map(category => (
            <GifGrid
              key={category}
              category={category}
            />
          ))
        }
      </center>

    </>
  )
}
