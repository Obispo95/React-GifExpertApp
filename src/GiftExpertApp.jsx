import { useState } from "react"
import { AddCategory, GiftGrid } from './components';

export const GiftExpertApp = () => {

  const [categories, setCategories] = useState(['Cats']);

  const onAddCategory = (newCategory) => {

    // setCategories(categories => [...categories, 'Valorant']);
    // setCategories(['Valorant', ...categories]);
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  }

  return (
    <>
      <center>
        <h1 className="title">GIF EXPERT APP</h1>

        <AddCategory
          onNewCategory={onAddCategory}
        />

        {
          categories.map(category => (
            <GiftGrid
              key={category}
              category={category}
            />
          ))
        }
      </center>

    </>
  )
}
