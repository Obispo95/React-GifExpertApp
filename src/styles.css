* {
    font-family: Helvetica, Arial, sans-serif;
  }
  
  body {
    padding: 70px;
    background-image: url('./image/bgcover.jpg');
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-position: center;
    background-size: cover;
    padding: 0px 15%;
  }

  input {
    background-color: white;
    border-radius: 5px;
    border: 2px solid rgb(97, 32, 158);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 50%;
  }
  
  h2 {
    font-size: 1.5rem;
  }
  
  h3 {
    background-color: white;
    border-radius: 20px 20px 0px 0px;
    border-top: 2px solid rgb(97, 32, 158);
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.07);
    font-size: 3rem;
    margin-bottom: 0;
    margin-top: 10px ;
    padding: 20px;
  }
  
  .title{
    color: darkblue;
    font-size: 3rem;
    border-radius: 10px;
  }

  .card-grid {
    margin: 0px;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    background-color: white;
    border-radius: 0px 0px 20px 20px;
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.07);
    border-bottom: 2px solid rgb(97, 32, 158);
  }
  
  .card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
    display: flex;
    flex-direction: column;
    height: 0%;  
    justify-content: center;
    margin: 2% 1% 2% 1%;
    overflow: hidden;
  }
  
  .card p {
    background-color: white;
    flex: 1;
    font-size: 1.2rem;
    margin-top: 5px;
    padding: 5px 10px 0px 10px;
    text-align: center;
  }
  
  .card img {
    width: 100%;
  }