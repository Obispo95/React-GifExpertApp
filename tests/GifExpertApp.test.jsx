import { render,screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('Pruebas en el componente <GiftExpertApp />', () => { 
    
    test('Identica al snapshot', () => { 
    
        const {container}= render (<GifExpertApp/>)

        expect(container).toMatchSnapshot();

     })

 })