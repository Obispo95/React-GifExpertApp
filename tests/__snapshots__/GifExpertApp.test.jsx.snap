// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GiftExpertApp /> Identica al snapshot 1`] = `
<div>
  <center>
    <h1
      class="title"
    >
      GIF EXPERT APP
    </h1>
    <form
      aria-label="form"
    >
      <input
        placeholder="Buscar Gifs"
        type="text"
        value=""
      />
    </form>
    <h3>
      Cats
    </h3>
    <h2>
      Loading...
    </h2>
    <div
      class="card-grid"
    />
  </center>
</div>
`;
